### DISCUSSION TITLE
Why is webpack bundling everything in the entire project

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#14001] [Q&A]
I'm perhaps just an idiot but I don't know what to try any more and I can't figure out why webpack keeps bundling everything in the entire project.  

This is my config:

````
import path from 'path'
import MiniCssExtractPlugin from 'mini-css-extract-plugin'
import TerserWebpackPlugin from 'terser-webpack-plugin'
import CssMinimizerWebpackPlugin from 'css-minimizer-webpack-plugin'

import type {Configuration, Compiler} from 'webpack'

export const config: Configuration = {
  mode: 'production',
  entry: path.join(__dirname, '..', 'src/ui/index.tsx'),
  output: {
    path: path.join(__dirname, '..', 'public'),
    filename: 'script.js',
    assetModuleFilename: '[name][ext]',
    publicPath: '/',
    libraryTarget: 'umd',
    clean: false,
  },
  target: 'web',
  resolve: {
    extensions: ['.js', '.ts', '.jsx', '.tsx'],
  },
  module: {
    rules: [
      {
        test: /.tsx?$/,
        use: 'ts-loader',
        exclude: /node_modules/,
      },
      {
        test: /\.(c|sc|sa)ss$/,
        use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
        exclude: /node_modules/,
      },
      {
        test: /\.(ico|svg|png|jpg)$/,
        type: 'asset/resource',
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: 'style.css',
    }),
  ],
  stats: 'errors-warnings',
  performance: {
    hints: 'error',
    maxAssetSize: 400000,
    maxEntrypointSize: 400000,
  },
  optimization: {
    minimize: true,
    minimizer: [
      (compiler: Compiler): void => {
        new TerserWebpackPlugin({
          parallel: true,
          extractComments: false,
        }).apply(compiler)
      },
      new CssMinimizerWebpackPlugin(),
    ],
  },
}

export default config
````

I've tried to change the rules, I've tried to change the resolves, I've tried to change the mode, but no matter what webpack goes crazy on bundling stuff, it even bundles jest and unit tests which is absurd.  

What's more is that:

````
import React from 'react'
import {render} from 'react-dom'

import HTML from './html'

const UI = () => {
  return <HTML />
}

render(<UI />, document.getElementById('ui'))
````

Along with HTML which has barely anything in it is the only 2 components I have, both of which don't import anything other than react.  If I coded this in normal HTML it should be a couple of KB big, but instead it's creating a 44k line script file that's literally breaking my IDE from the amount of junk in there.  

What config option am I missing or what's causing webpack to bundle everything in the entire project.  Intuitively I thought that webpack should only start at the entry and include everything that's imported from there, but it seems like it's just bundling everything no matter what.  Am I doing something stupid.

**Comment by ludofischer** [comment]
>  Intuitively I thought that webpack should only start at the entry and include everything that's imported from there

That's how it usually works. I can't see anything wrong with your config on first inspection, have you tried using the `context` option to specify the file paths (https://webpack.js.org/configuration/entry-context/#context)? It's hard to figure out what's going on without seeing the directory layout. What seems slightly unusual is that you seem to be running webpack one level deep inside the directory that contains the source code (`entry: path.join(__dirname, '..', 'src/ui/index.tsx')`).
